src/Setup/DevDivInsertionFiles/BuildDevDivInsertionFiles.vb

```diff
@@ -911,6 +911,8 @@ Public Class BuildDevDivInsertionFiles
                           <version>0.0</version>
                       </metadata>
                       <files>
+                          <file src="System.Collections.Immutable.dll"/>
```

Comment:
Why not update whoever is calling GenerateRoslynCompilerNuSpec to .Concat() the filenames into the list? That way you have the dedup logic and whatever else might be needed too.